xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp

```diff
@@ -82,7 +82,12 @@ bool CInputStreamAddon::Supports(BinaryAddonBasePtr& addonBase, const CFileItem
     return (addon.asString() == addonBase->ID());
 
   // check protocols
-  std::string protocol = fileitem.GetURL().GetProtocol();
+  std::string protocol;
+  if (!fileitem.GetDynPath().empty())
```

Comment:
no need to test for empty. GetDynPath already  returns GetPath if dynPath is empty